src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentRequestMapper.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.request.PaymentRequest;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.config.SiteConfiguration;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.exception.SiteNotFoundException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.model.PaymentMessage;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@EnableConfigurationProperties(SiteConfiguration.class)
+public class PaymentRequestMapper {
+
+    private final SiteConfiguration siteConfiguration;
+
+    public PaymentRequestMapper(SiteConfiguration siteConfiguration) {
+        this.siteConfiguration = siteConfiguration;
+    }
+
+    public PaymentRequest mapPaymentMessage(PaymentMessage message) {
+        return new PaymentRequest(
+            message.ccdCaseNumber,
+            getPaymentDocumentControlNumbers(message),
+            message.isExceptionRecord,
+            getSiteIdForPostCode(message.poBox)
+        );
+    }
+
+    private List<String> getPaymentDocumentControlNumbers(PaymentMessage message) {
```

Comment:
I wonder if there should be something that fully validates the message first and here we could just assume the message is valid.

Reply:
good point we could do that as a separate PR though